### PR TITLE
add spelling check script

### LINKED ISSUE #8508: add spelling check script
**Describe the feature request**
Recently, saw many  PR about `fix typo`, we should better add spelling check script in `/bin` directory.
So CI can check typo. I'll have a try
/assign

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by stewart-yu**
add spelling check script
fix #8508

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
Glad to see that. Thanks @stewart-yu

**Comment by hzxuzhonghu** [comment]
@stewart-yu  I prefer a separate commit only include Gopkg.toml modification. And then second commit : run `dep ensure` to add vendor.

File: mixer/adapter/list/config/adapter.list.config.pb.html [hunk 1]
```diff
@@ -32,7 +32,7 @@ <h2 id="Params">Params</h2>
 <td><code>providerUrl</code></td>
 <td><code>string</code></td>
 <td>
-<p>Where to find the list to check against. This may be ommited for a completely local list.</p>
+<p>Where to find the list to check against. This may be omitted for a completely local list.</p>
```
Comment (geeknoid) [inline mixer/adapter/list/config/adapter.list.config.pb.html]: Fixes in the various .pb.html files need to be reflected in the original .proto files. the .pb.html rile is auto-generated from the .proto.

Comment (stewart-yu) [reply mixer/adapter/list/config/adapter.list.config.pb.html]: sorry, little knowledge about it:wink:
In this condition, i would like add `.pb.html` to `bin/.spelling_failure` file. If i wrong, please correct me

Comment (stewart-yu) [reply mixer/adapter/list/config/adapter.list.config.pb.html]: Done, fixed in last commit. thanks

**Pull request merged**